dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/DicomImageReader.java

```diff
@@ -394,63 +236,64 @@ public boolean canReadRaster() {
         return true;
     }
 
+
     @Override
     public Raster readRaster(int frameIndex, ImageReadParam param)
             throws IOException {
         readMetadata();
         checkIndex(frameIndex);
 
-        openiis();
-        try {
+        // Need to map from Fragment to IIS.
+        Raster raster;
+        try (ImageInputStream iisOfFrame = iisOfFrame( frameIndex)) {
```

Comment:
This is MUCH cleaner doing the open/close around this and not worrying about the underlying state.

Reply:
Thanks.  It is also friendly for IDEs which like to detect resource leaks.